contracts/accessories/listener/src/contract.rs

```diff
@@ -0,0 +1,71 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg, ResultResponse};
+use crate::state::{NOTE, RESULTS};
+
+const CONTRACT_NAME: &str = "crates.io:polytone-listener";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let note = deps.api.addr_validate(&msg.note)?;
+    NOTE.save(deps.storage, &note)?;
+
+    Ok(Response::default()
+        .add_attribute("method", "instantiate")
+        .add_attribute("note", msg.note))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    // Only the note can call this contract.
+    if info.sender != NOTE.load(deps.storage)? {
+        return Err(ContractError::Unauthorized {});
+    }
```

Comment:
Do you really want only the note to call this contract? 
I would enter this check inside the callback msg because:

1. We might want some other contracts to call this contract, or have the listener call it self for something in the future
2. Note will only ever call the callback msg anyways, other msgs will probably will have other "admins"

Reply:
good idea thank you!! 🙏 

moved it to the callback message specifically